alf/algorithms/mbrl_algorithm.py

```diff
@@ -174,8 +174,8 @@ def _calc_step_reward(self, obs, action, state: MbrlState):
     def _predict_with_planning(self, time_step: TimeStep, state,
                                epsilon_greedy):
         # full state in
-        action = self._planner_module.generate_plan(time_step, state,
-                                                    epsilon_greedy)
+        action, state = self._planner_module.predict_plan(
+            time_step, state, epsilon_greedy)
```

Comment:
For the planner module, `state.planner` is different from `state.dynamics` or `state.reward`. `state.dynamics` and `state.reward` are more like an input. While `state.planner` is indeed a state.

Reply:
I have refactored the APIs for better resolving the issue.